lab_6_pipeline/pipeline.py

```diff
@@ -32,16 +53,50 @@ def __init__(self, path_to_raw_txt_data: pathlib.Path) -> None:
         Args:
             path_to_raw_txt_data (pathlib.Path): Path to raw txt data
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validate folder with assets.
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError('No such file.')
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError('Path does not lead to directory.')
+        if not next(self.path_to_raw_txt_data.iterdir(), None):
+            raise EmptyDirectoryError('Empty directory.')
+        meta_actual = []
+        raw_actual = []
+        for file in self.path_to_raw_txt_data.iterdir():
+            if '_raw.txt' in file.name:
+                raw_actual.append(file.name)
+            elif '_meta.json' in file.name:
+                meta_actual.append(file.name)
+            if not file.stat().st_size:
+                raise InconsistentDatasetError('Empty file(s) in the directory.')
+        if len(meta_actual) != len(raw_actual):
+            raise InconsistentDatasetError('Number of raw and meta files is not equal.')
+        raw_ideal = [f'{num}_raw.txt' for num in range(1, len(raw_actual) + 1)]
+        meta_ideal = [f'{num}_meta.json' for num in range(1, len(meta_actual) + 1)]
+        if set(meta_actual) != set(meta_ideal) or set(raw_actual) != set(raw_ideal):
```

Comment:
you also need to check that there are no slips in numbering of files, like: 1, 2, 4, 5, 6 with 3 being missed